lib/idempotence/reservation/reservation.rb

```diff
@@ -108,11 +108,15 @@ class Reservation
         attr_reader :message
         attr_reader :idempotence_key
 
-        def call(m, i_key, &block)
-          @message = m
-          @idempotence_key = i_key
+        def call(message, idempotence_key, &block)
+          @message = message
+          @idempotence_key = idempotence_key
 
-          yield message
+          yield message if @yield
+        end
+
+        def set_reserved(value)
+          @yield = value
         end
```

Comment:
just a suggestion: maybe instead of `@yield` we should have `@process_block`. It'll make it easier to understand that setting reserved means run/process block.

Reply:
done